src/service/url-replacements-impl.js

```diff
@@ -594,6 +607,28 @@ export class GlobalVariableSource extends VariableSource {
       return getter(/** @type {!ShareTrackingFragmentsDef} */ (fragments));
     });
   }
+
+  /**
+   * Resolves the value via amp-story's service.
+   * @param {function(!{pageIndex: number, pageId: string}):T} getter
+   * @param {string} expr
+   * @return {!Promise<T>}
+   * @template T
+   * @private
+   */
+  getStoryValue_(getter, expr) {
+    if (!this.storyVariables_) {
+      this.storyVariables_ =
+          Services.storyVariableServiceForOrNull(this.ampdoc.win);
```

Comment:
This could return `null` again (if the AMP page doesn't have the story extension installed), which means calling `.then()` on it will cause an error. In that case, let's throw a user error asking them to install amp-story.

Reply:
1. Re: error. I think the convention in this case is just to assert the error on promise resolution.
2. Re: null-value, I don't think that's the case. `storyVariableServiceForOrNull` will return a promise that resolves to null, not null itself.

See `getVairiantsValue_` (sic), `getShareTrackingValue_` for prior art on the same use-case (i.e. reading extension-specific values).

Reply:
Oh, my mistake. Can we change `this.storyVariables_` to `this.storyVariablesPromise_`?

Reply:
Sure, done.